pkg/api/types.go

```diff
@@ -1414,6 +1414,38 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	unversioned.TypeMeta `json:",inline"`
+	ObjectMeta           `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// The minimum number of pods that must be available simultaneously.  Exactly
```

Comment:
My reason here was that I didn't want to constantly be checking if it's an int or string, decoding the string into a number and multiplying.  I preferred a back-end representation that already had the percentage decoded, at which point it made more sense to have the public representation match the back-end.

If I switch to IntOrString, I guess all consumers of this info will just do this work over and over again.  It's not horrible inefficiency, but it feels wrong to be wasteful merely for consistency.  Perhaps if we want consistency we should revise Deployment to use this "exactly one" approach in v2?